app/assets/javascripts/discourse/app/components/composer-editor.js

```diff
@@ -218,34 +218,42 @@ export default Component.extend(ComposerUpload, {
     return extensions.map((ext) => `.${ext}`).join();
   },
 
+  @bind
+  _afterMentionComplete(value) {
+    this.composer.set("reply", value);
+
+    // ensures textarea scroll position is correct
+    schedule("afterRender", () => {
+      const input = this.element.querySelector(".d-editor-input");
+      input?.blur();
+      input?.focus();
+    });
+  },
+
   @on("didInsertElement")
   _composerEditorInit() {
     const $input = $(this.element.querySelector(".d-editor-input"));
-    const $preview = $(this.element.querySelector(".d-editor-preview-wrapper"));
 
     if (this.siteSettings.enable_mentions) {
       $input.autocomplete({
         template: findRawTemplate("user-selector-autocomplete"),
-        dataSource: (term) => this.userSearchTerm.call(this, term),
+        dataSource: this._userSearchTerm,
         key: "@",
         transformComplete: (v) => v.username || v.name,
-        afterComplete: (value) => {
-          this.composer.set("reply", value);
-
-          // ensures textarea scroll position is correct
-          schedule("afterRender", () => $input.blur().focus());
-        },
+        afterComplete: this._afterMentionComplete,
         triggerRule: (textarea) =>
           !inCodeBlock(textarea.value, caretPosition(textarea)),
       });
     }
 
     if (this._enableAdvancedEditorPreviewSync()) {
-      this._initInputPreviewSync($input, $preview);
+      const input = this.element.querySelector(".d-editor-input");
+      const preview = this.element.querySelector(".d-editor-preview-wrapper");
+      this._initInputPreviewSync(input, preview);
     } else {
-      $input.on("scroll", () =>
-        throttle(this, this._syncEditorAndPreviewScroll, $input, $preview, 20)
-      );
+      this.element
+        .querySelector(".d-editor-input")
+        ?.addEventListener("scroll", this._throttledSyncEditorAndPreviewScroll);
```

Comment:
Should this event listener be removed at some point? (e.g. `willDestroyElement`)

Reply:
Yes indeed, that should do it: https://github.com/discourse/discourse/pull/14924/commits/d016cd2b340c706cb662e2d9767bebecccf067cc